src/calendar.cpp

```diff
@@ -441,6 +441,17 @@ int calendar::day_of_year() const
     return day + season_length() * season;
 }
 
+int calendar::diurnal_time_before( int turn ) const
+{
+    const int remainder = turn - get_turn() % DAYS( 1 );
+    return ( remainder >= 0 ) ? remainder : DAYS( 1 ) + remainder;
+}
+
+int calendar::diurnal_time_before( const calendar &time ) const
```

Comment:
Why do you need this overload? `calendar` has a conversion to `int` operator. Calling the `diurnal_time_before` with a `calendar` object should just invoke the operator and use its result `turn`.